es-core/src/components/VideoVlcComponent.cpp

```diff
@@ -255,7 +255,8 @@ void VideoVlcComponent::startVideo()
 			{
 				unsigned track_count;
 				// Get the media metadata so we can find the aspect ratio
-				libvlc_media_parse(mMedia);
```

Comment:
Compier warning. Deprecated function call replaced with recommended function.

/home/devel/temp/EmulationStation_Stable/es-core/src/components/VideoVlcComponent.cpp:258:5: warning:
 'libvlc_media_parse' is deprecated [-Wdeprecated-declarations]
                                libvlc_media_parse(mMedia);
                                ^
/usr/include/vlc/deprecated.h:208:1: note: 'libvlc_media_parse' has been explicitly marked deprecated
 here
LIBVLC_DEPRECATED LIBVLC_API void
^
/usr/include/vlc/libvlc.h:51:43: note: expanded from macro 'LIBVLC_DEPRECATED'
# define LIBVLC_DEPRECATED __attribute__((deprecated))
                                          ^
1 warning generated.

Reply:
Dang it! I would suggest basing it on:
```
/** LibVLC full version as a single integer (for comparison) */
# define LIBVLC_VERSION_INT \
         LIBVLC_VERSION(LIBVLC_VERSION_MAJOR, LIBVLC_VERSION_MINOR, \
                        LIBVLC_VERSION_REVISION, LIBVLC_VERSION_EXTRA)

```
from libvlc_version.h, do you know what version of libvlc the pi's are using?

Reply:
Very good question.

On the Pi I'm testing compilation for, this is what I'm reading:

```
libvlc-bin/oldstable 3.0.11-0+deb9u1 armhf [upgradable from: 2.2.6-1~deb9u1+rpi1]
libvlc-dev/oldstable 3.0.11-0+deb9u1 armhf [upgradable from: 2.2.6-1~deb9u1+rpi1]
libvlc5/oldstable 3.0.11-0+deb9u1 armhf [upgradable from: 2.2.6-1~deb9u1+rpi1]
libvlccore-dev/oldstable 3.0.11-0+deb9u1 armhf [upgradable from: 2.2.6-1~deb9u1+rpi1]
libvlccore8/now 2.2.6-1~deb9u1+rpi1 armhf [installed,local]
libvlccore9/oldstable 3.0.11-0+deb9u1 armhf

```


On a Pi I have that I think I have the latest OS version on, it's this:

```
libvlc-bin-dbgsym/oldstable 3.0.17.4-0+deb10u2+rpt1 arm64
libvlc-bin-dbgsym/oldstable 3.0.17.4-0+deb10u2+rpt1 armhf
libvlc-bin/oldstable 3.0.17.4-0+deb10u2+rpt1 arm64
libvlc-bin/oldstable 3.0.17.4-0+deb10u2+rpt1 armhf
libvlc-dev/oldstable 3.0.17.4-0+deb10u2+rpt1 arm64
libvlc-dev/oldstable 3.0.17.4-0+deb10u2+rpt1 armhf
libvlc5-dbgsym/oldstable 3.0.17.4-0+deb10u2+rpt1 arm64
libvlc5-dbgsym/oldstable 3.0.17.4-0+deb10u2+rpt1 armhf
libvlc5/oldstable 3.0.17.4-0+deb10u2+rpt1 arm64
libvlc5/oldstable 3.0.17.4-0+deb10u2+rpt1 armhf
libvlccore-dev/oldstable 3.0.17.4-0+deb10u2+rpt1 arm64
libvlccore-dev/oldstable 3.0.17.4-0+deb10u2+rpt1 armhf
libvlccore9-dbgsym/oldstable 3.0.17.4-0+deb10u2+rpt1 arm64
libvlccore9-dbgsym/oldstable 3.0.17.4-0+deb10u2+rpt1 armhf
libvlccore9/oldstable 3.0.17.4-0+deb10u2+rpt1 arm64
libvlccore9/oldstable 3.0.17.4-0+deb10u2+rpt1 armhf
```

Does this help? What's on your Ubuntu machine?

Reply:
Meaning, it seems I'm on 2, but it seems 3 is available. I have not tested building on 3 though.

Reply:
Ubuntu 22.04.2 used 3.0.16.0 so this approach would resolve your immediate compilation issue. But it would also need to be tested in your Pi with the latest OS using 3.0.17.4.

I can push a change for testing...

Reply:
From what I can tell it was added in 3.0.0

https://git.m2x.eu/videolan/vlc/-/blob/master/NEWS

Reply:
I'd be supportive of using 3.0 as the version, then, and using the different methods accordingly. I'm checking with others if they'd be ok with that, or if it's actually overkill and we can just assume users could install libvlc-3.0. Or maybe we'd need to update the dependencies list somewhere along with this PR. So, if you'd want to wait for a bit I'll get back to you here.

Apologies for the trouble here, and thanks for enduring it! :)

Reply:
No problem. Better to take some time to get to the right decision. Let me know who you want to proceed.
IH.

Reply:
I think it's ok to bump the `libVLC` version to 3, it's been included in Debian since Debian 10 _buster_ (2019). We may want to bump the version check in https://github.com/RetroPie/EmulationStation/blob/f4c3815b44d14593c8708c116d1678850afd04b0/CMake/Packages/FindVLC.cmake#L26 to ensure we're not using an older version and we can document the version requirement in a separate commit.

I have some un-related comments after testing the PR, but I'll add them separately.